### ISSUE TITLE
Question about super-resolution losses

### ISSUE THREAD

**Issue by faroit** [j-rixen/SFSRNet-Audio-samples#1]
Hi @j-rixen, thanks for this great paper. I have a question regarding the loss function you describe in the paper when using to sr any backbone model:

![image](https://user-images.githubusercontent.com/72940/216303465-da4f3e07-71f6-4603-928e-49a721ee22e8.png)

I do understand what you mean by the second and third loss. But i fail to understand the first one.
Here is an example:

```
    def forward(self, mix: torch.Tensor) -> dict[str, torch.Tensor]:
        downsampled_input = self.downsample(mix)
        X = self.encoder(downsampled_input)
        Q = self.separator(X)
        S = self.masker(Q) * X
        y_downsampled = self.decoder(S)
        y = self.upsample(y_downsampled)
        y_sr = self.sr_net(mix, y)

```

- loss 2 is "optimizing the final estimations of the separation": `criterion(y, target)`
- loss 3 is "optimizing the estimations that are calculated before the SR network: `criterion(y, downsampled_target)`
- loss 1 is "optimizing the SR network": I don't know what you mean by this. Would this be a separate training to upsample the targets? e.g. `criterion(sr_net(input_signal, downsampled_target), target)`?
